android/src/main/java/com/amplitude/android/plugins/AndroidLifecyclePlugin.kt

```diff
@@ -2,35 +2,78 @@ package com.amplitude.android.plugins
 
 import android.app.Activity
 import android.app.Application
+import android.content.pm.PackageInfo
+import android.content.pm.PackageManager
 import android.os.Bundle
 import com.amplitude.android.Configuration
+import com.amplitude.android.utilities.DefaultEventUtils
 import com.amplitude.core.Amplitude
 import com.amplitude.core.platform.Plugin
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
 
 class AndroidLifecyclePlugin : Application.ActivityLifecycleCallbacks, Plugin {
     override val type: Plugin.Type = Plugin.Type.Utility
     override lateinit var amplitude: Amplitude
+    private lateinit var packageInfo: PackageInfo
+
+    private val hasTrackedApplicationLifecycleEvents = AtomicBoolean(false)
+    private val numberOfActivities = AtomicInteger(1)
+    private val isFirstLaunch = AtomicBoolean(false)
 
     override fun setup(amplitude: Amplitude) {
         super.setup(amplitude)
-        ((amplitude.configuration as Configuration).context as Application).registerActivityLifecycleCallbacks(this)
+
+        val application = (amplitude.configuration as Configuration).context as Application
+        val packageManager: PackageManager = application.packageManager
+        packageInfo = try {
+            packageManager.getPackageInfo(application.packageName, 0)
+        } catch (e: PackageManager.NameNotFoundException) {
+            throw AssertionError("Cannot find package with application.packageName: " + application.packageName)
+        }
+        application.registerActivityLifecycleCallbacks(this)
     }
 
     override fun onActivityCreated(activity: Activity, bundle: Bundle?) {
+        val configuration = amplitude.configuration as Configuration
+        if (!hasTrackedApplicationLifecycleEvents.getAndSet(true) && configuration.trackingAppLifecycleEvents) {
+            numberOfActivities.set(0)
+            isFirstLaunch.set(true)
+            DefaultEventUtils(amplitude as com.amplitude.android.Amplitude).trackAppUpdatedInstalledEvent(packageInfo)
```

Comment:
would this be triggered on each time app open?

Reply:
`trackAppUpdatedInstalledEvent` will not send event each time app opened:
- for installed event, it will check previous build, if it has (after first time installed), then skip sending installed event
- for updated event, it will compare current build to previous build, if they are the same (no app build upgraded), then skip sending updated event